test/core/stake/closeMetaBlock.js

```diff
@@ -0,0 +1,101 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const BN = require('bn.js');
+const Events = require('../../test_lib/event_decoder.js');
+const StakeUtils = require('./helpers/stake_utils.js');
+const Utils = require('../../test_lib/utils.js');
+
+const MockToken = artifacts.require('MockToken');
+const Stake = artifacts.require('Stake');
+
+contract('Stake.closeMetaBlock()', async (accounts) => {
+
+    let originCoreAccount = accounts[1];
+    let minimumWeight = new BN('1');
+    let token;
+    let stake;
+
+    beforeEach(async () => {
+        token = await MockToken.new();
+        stake = await Stake.new(
+            token.address,
+            originCoreAccount,
+            minimumWeight,
+        );
+        await StakeUtils.initializeStake(
+            stake,
+            token,
+            accounts[0],
+            [accounts[2]],
+            [new BN('2')],
+        );
+    });
+
+    it('should increase the meta-block height by 1', async () => {
+        for (let expectedHeight = 1; expectedHeight < 5; expectedHeight++) {
+            await stake.closeMetaBlock(
+                new BN(expectedHeight - 1),
+                {from: originCoreAccount}
+            );
+
+            height = await stake.height.call();
+            assert.strictEqual(
+                expectedHeight,
+                height.toNumber(),
+                "The height should increase by one when an meta-block is closed."
+            );
+        }
+    });
+
+    it('should emit an event when an meta-block is closed', async () => {
+        let tx = await stake.closeMetaBlock(
+            new BN(0),
+            {from: originCoreAccount}
+        );
+        let events = Events.perform(tx.receipt, stake.address, stake.abi)
+        assert.strictEqual(
+            Number(events.HeightIncreased.newHeight),
+            1,
+            'The contract did not emit an event with the new height.'
+        );
+
+        tx = await stake.closeMetaBlock(
+            new BN(1),
+            {from: originCoreAccount}
+        );
+        events = Events.perform(tx.receipt, stake.address, stake.abi)
+        assert.strictEqual(
+            Number(events.HeightIncreased.newHeight),
+            2,
+            'The contract did not emit an event with the new height.'
+        );
+    });
+
+    it('should fail when a wrong height is given', async () => {
+        await Utils.expectFailedAssert(
+            stake.closeMetaBlock(
+                new BN(3),
+                {from: originCoreAccount}
+            )
+        );
```

Comment:
Why we are using `assert` and not `require` in the contract?
Being an external function and `_closingHeight` as param, we can expect wrong height.
We use `assert` when we know things have unexpectedly gone wrong.

Reply:
`closeMetaBlock()` can only be called by the origin core. If the assert fails, there must be a bug in the origin core contract or in the integration of the contracts.

Reply:
We can keep `assert` for now, we can check if it should be `require` or `assert` when origin core implements the calling of `closeMetaBlock`.